docs/se/openapi-geneator.adoc

```diff
@@ -0,0 +1,218 @@
+OpenAPI documents allow you to convert the metadata about your API into some other format and OpenAPI Generator helps with that. It allows generating of API client or server stubs.
+With OpenAPI client generator for Helidon SE, user can easily generate Rest Client to interact with any server which exposes an OpenAPI document.
+With OpenAPI server generator for Helidon SE server stubs can be generated where user has to implement business logic for exposed endpoints.
+
+== Maven Coordinates [[Maven-Coordinates]]
+
+Core:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.openapitools</groupId>
+    <artifactId>openapi-generator</artifactId>
+    <version>${openapi-generator-version}</version>
+</dependency>
+----
+
+See the different versions of the openapi-generator artifact available on maven central.
+
+Cli:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.openapitools</groupId>
+    <artifactId>openapi-generator-cli</artifactId>
+    <version>${openapi-generator-version}</version>
+</dependency>
+----
+
+See the different versions of the openapi-generator-cli artifact available on maven central.
+
+Maven plugin:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.openapitools</groupId>
+    <artifactId>openapi-generator-maven-plugin</artifactId>
+    <version>${openapi-generator-version}</version>
+</dependency>
+----
+
+== Configuration
+
+To customize generated project different options can be used.
+Options for OpenAPI Generator are the same whether you're using the Maven/Gradle Plugins, CLI, or Online generation.
+
+General Configuration parameters
+
+At a minimum generation of the new Helidon MP project requires:
+
+[cols="1,1"]
+|===
+|Option | Description
+
+|inputSpec
+|OpenAPI Spec file path
+|generatorName
+|target generator name
+|library
+|library template (sub-template)
+|===
+
+The whole list of the general configuration parameters for OpenAPI Generator is available on link:https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin[openapi-generator-maven-plugin] page.
+
+Predefined values for required options for Helidon SE Generators :
+
+[cols="1,1,1"]
+|===
+|Option |Value |Description
+
+|generatorName
+|java-helidon-server
+|generator name for Helidon SE Server
+
+|generatorName
+|java-helidon-client
+|generator name for Helidon SE Client
+
+|library
+|se
+|used library for Helidon SE Server/Client
+|===
+
+OpenAPI Generators for Helidon SE support such specific options :
+
+[cols="1,1,1,1,1"]
+|===
+|Option |Description |Values |Default |Notes
+
+|helidonVersion
+|Version of Helidon
+|
+|2.5.2
+|
+
+|fullProject
+|Generate API files only
+|true/false
+|false
+| API files : API Interfaces or abstract classes, models (files we do not expect the end user to modify)
+
+|serializationLibrary
+|Library to use for serializing JSON
+|jsonb, jackson
+|jackson
+|
+|===
+
+== Usage
+
+CLI, Maven/Gradle Plugins, or Online generation can be used to generate Helidon SE project.
```

Comment:
You can use the CLI, Maven/Gradle plugins, or online generation to generate a Heldion SE project.